src/gfx/shaders/VolumeMaterial.js

```diff
@@ -35,43 +35,46 @@ function overrideUniforms(params, defUniforms) {
   return uniforms;
 }
 
-function BackFacePosMaterial(params) {
-  const settings = {
-    uniforms: overrideUniforms(params, {}),
-    vertexShader: `\
+class BackFacePosMaterial extends THREE.ShaderMaterial {
```

Comment:
I do agree with SonarCloud and CodeClimate that in addition to wrapping this in ES2015 classes we should get rid of the duplicated code here. Please check the reports and try to generalize the behavior. You could also put shaders in separate glsl files as we wanted to (there's a separate task for that in the whole project but you "touch" this particular code anyway).